cloudProvider.go

```diff
@@ -0,0 +1,81 @@
+package firewall
+
+import (
+	_ "embed"
+	"encoding/json"
+)
+
+var (
+	//go:embed cloudProviderData/aws-ip-ranges.json
+	awsFile []byte
+
+	//go:embed cloudProviderData/azure-ip-ranges.json
+	azureFile []byte
+	//go:embed cloudProviderData/gcp-ip-ranges.json
+	gcpFile []byte
+)
+
+
+type awsRanges struct {
+	Prefixes []awsPrefix `json:"prefixes"`
+}
+
+type awsPrefix struct {
+	IPPrefix   string `json:"ip_prefix"`
+	IPv6Prefix string `json:"ipv6_prefix"`
+}
+
+type azureRanges struct {
+	Prefixes []azurePrefix `json:"prefixes"`
+}
+
+type azurePrefix struct {
+	IPPrefix string `json:"ip_prefix"`
+}
+
+type gcpRanges struct {
+	Prefixes []gcpPrefix `json:"prefixes"`
+}
+
+type gcpPrefix struct {
+	IPPrefix   string `json:"ipv4Prefix"`
+	IPv6Prefix string `json:"ipv6Prefix"`
+}
+
+// CloudProviderBlockList returns a slice of IP Ranges of aws, azure and gcp
+func CloudProviderBlockList() (blockList []string) {
```

Comment:
you're doing this per-call, but this should be calculated once and memoized.

either add a memoization variable.. ie. `var cloudProviderBlockList []string = nil` as package-level then return that early if its set, otherwise compute + save.. or do this inside of `init()`

Reply:
So this should be changed in _example/basic/main.go right?